lib/mp4/probe.js

```diff
@@ -221,10 +221,106 @@ getVideoTrackIds = function(init) {
   return videoTrackIds;
 };
 
+getTracks = function(init) {
+  var traks = findBox(init, ['moov', 'trak']);
+  var tracks = [];
+
+  traks.forEach(function(trak) {
+    var track = {};
+    var tkhd = findBox(trak, ['tkhd'])[0];
+
+    // id
+    if (tkhd) {
+      var view = new DataView(tkhd.buffer, tkhd.byteOffset, tkhd.byteLength);
+      var version = view.getUint8(0);
+
+      track.id = (version === 0) ? view.getUint32(12) : view.getUint32(20);
+    }
+
+    var hdlr = findBox(trak, ['mdia', 'hdlr'])[0];
+
+    // type
+    if (hdlr) {
+      var type = parseType(hdlr.subarray(8, 12));
+
+      if (type === 'vide') {
+        track.type = 'video';
+      } else if (type === 'soun') {
+        track.type = 'audio';
+      } else {
+        track.type = type;
+      }
+    }
+
+
+    // codec
+    var stsd = findBox(trak, ['mdia', 'minf', 'stbl', 'stsd'])[0];
+
+    if (stsd) {
+      var sampleDescriptions = stsd.subarray(8);
+      // gives the codec type string
+      track.codec = parseType(sampleDescriptions.subarray(4, 8));
+
+      // https://tools.ietf.org/html/rfc6381#section-3.3
+      if ((/^[a-z]vc[1-9]$/i).test(track.codec)) {
+        var codecConfig = sampleDescriptions.subarray(94);
+        // always avcc
+        // var type = parseType(codecConfig.subarray(4, 8));
```

Comment:
what's happening here? Also why `subarray(94)`?

Reply:
If we find that we are dealing with an `avc1` codec we skip over the `avc1` box, as it doesn't contain anything relevant to parsing the codec parameters, and move to the next box. Which appears to be `avcC`. That box starts at `subarray(94)`. Perhaps we should use `findBox` here, so that it makes more sense.

Reply:
Yeah, I think a combination of using `findBox` and comments would help here